Auto-reconnect to last connected Bluetooth device

qwertychouskie opened the pull request:
Stores the BT address in flash memory, so auto-connection can work across power cycles.  Only writes address to flash if it differs from the previously stored address, to avoid unnecessary flash wear.

qwertychouskie committed:
Auto-reconnect to last connected Bluetooth device

Stores the BT address in flash memory, so auto-connection can work across power cycles.  Only writes address to flash if it differs from the previously stored address, to avoid unnecessary flash wear.

qwertychouskie committed:
Expose disconnect() and reconnect() functions

For some reason, reconnecting won't work after a full de-init, at least until the ESP32 is power cycled, so allow direct access to the disconnect and reconnect functions.

qwertychouskie committed:
Update README to document auto-connection

tierneytim commented:
This seems like a cool feature. Will test and integrate over the weekend

qwertychouskie commented:
@tierneytim Just wanted to check if you've had a chance to look at the code.  I have some local commits to my project (https://github.com/qwertychouskie/GM-EandC-Cemu-v2) that I want to push, but was going to wait until the library changes were merged first.
Thanks for your work on this project!

tierneytim commented:
Unfortunately I have not been well recently so I haven't had a chance to review but will get to it before Friday

tierneytim reviewed (changes requested):
Ok so I have had a look and think it's a great feature but I'm not sure embedding it in the begin method is the right way forward (as that makes it the default behavior beyond user control). 

Could you write two separate methods? One would check for an address in the object and save it if it was there. The other method would attempt a connection to whatever is currently saved(maybe in the reconnect method). Then users can choose whether to have this functionality or not?

Does that make sense?

qwertychouskie committed:
Move auto-reconnect code to reconnect()

This allows the user to decide whether to reconnect or not, rather than forcing the behavior.

qwertychouskie committed:
Update examples to include reconnect()

Also formatting and consistency improvements

qwertychouskie committed:
Update README.md with reconnect() info, formatting/typo fixes

qwertychouskie commented:
I made some changes and tested, everything still works in my project.  Should be ready for review again.

tierneytim reviewed (approved):
great work

qwertychouskie commented:
Thanks for the merge!
